### PR TITLE
add unicode emoji v11.0 provider

### PR DESCRIPTION
**Pull request by joke2k**
### What does this changes

Add new provider with a lot unicode emoji generator:
- emoji(category: name or list of names, keyword: term or list of terms)
- smile(): 😂 is a shortcut for emoji('face-positive')

Then there are one `emoji_*()` generator for each categories (and sub categories):

- fake.emoji_activities() ⚽
- fake.emoji_alphanum() 🈺
- fake.emoji_animal_amphibian() 🐸
- fake.emoji_animal_bird() 🐔
- fake.emoji_animal_bug() 🐝
- fake.emoji_animal_mammal() 🐐
- fake.emoji_animal_marine() \U0001f99e
- fake.emoji_animal_reptile() 🐍
- fake.emoji_animals_nature() 🦅
- fake.emoji_arrow() ⬆
- fake.emoji_arts & crafts() 🖼
- fake.emoji_av_symbol() 🔀
- fake.emoji_award_medal() 🎖
- fake.emoji_body() 👈
- fake.emoji_book_paper() 📑
- fake.emoji_cat_face() 😼
- fake.emoji_clothing() \U0001f9e5
- fake.emoji_computer() 🖲
- fake.emoji_country_flag() 🇨 🇬
- fake.emoji_dishware() 🍴
- fake.emoji_drink() 🍸
- fake.emoji_emotion() 💚
- fake.emoji_event() 🎋
- fake.emoji_face_fantasy() 👹
- fake.emoji_face_negative() 😞
- fake.emoji_face_neutral() 🤔
- fake.emoji_face_positive() 😆
- fake.emoji_face_role() \U0001f974
- fake.emoji_face_sick() 🤒
- fake.emoji_family() 👩 \u200d 👧
- fake.emoji_flag() 🎌
- fake.emoji_flags() 🇷 🇸
- fake.emoji_food_asian() 🍝
- fake.emoji_food_drink() \U0001f963
- fake.emoji_food_fruit() 🍋
- fake.emoji_food_prepared() \U0001f96f
- fake.emoji_food_sweet() \U0001f9c1
- fake.emoji_food_vegetable() 🥒
- fake.emoji_game() \U0001f9f8
- fake.emoji_geometric() ◾
- fake.emoji_hair_style() \U0001f9b2
- fake.emoji_hotel() 🛎
- fake.emoji_household() 🛁
- fake.emoji_keycap() 0 ️ ⃣
- fake.emoji_light & video() 🎬
- fake.emoji_lock() 🔒
- fake.emoji_mail() ✉
- fake.emoji_medical() 💊
- fake.emoji_money() 💳
- fake.emoji_monkey_face() 🙉
- fake.emoji_music() 📻
- fake.emoji_musical_instrument() 🎻
- fake.emoji_objects() 📠
- fake.emoji_office() 🗒
- fake.emoji_other_object() ⚰
- fake.emoji_other_symbol() 🔰
- fake.emoji_person() 👧
- fake.emoji_person_activity() 🕴
- fake.emoji_person_fantasy() \U0001f9de \u200d ♂ ️
- fake.emoji_person_gesture() 🙎 \u200d ♂ ️
- fake.emoji_person_role() 👩 \u200d 🎨
- fake.emoji_person_sport() 🏊 \u200d ♂ ️
- fake.emoji_phone() 📞
- fake.emoji_place_building() 🏪
- fake.emoji_place_geographic() 🏝
- fake.emoji_place_map() 🌐
- fake.emoji_place_other() 🌅
- fake.emoji_place_religious() ⛩
- fake.emoji_plant_flower() 💐
- fake.emoji_plant_other() 🌿
- fake.emoji_religion() ⚛
- fake.emoji_science() 📡
- fake.emoji_sky & weather() 🌞
- fake.emoji_smileys_people() 👋
- fake.emoji_sound() 🔔
- fake.emoji_sport() 🏐
- fake.emoji_subdivision_flag() 🏴 \U000e0067 \U000e0062 \U000e0073 \U000e0063 \U000e0074 \U000e007f
- fake.emoji_symbols() ↩
- fake.emoji_time() 🕞
- fake.emoji_tool() ⚒
- fake.emoji_transport_air() 🚟
- fake.emoji_transport_ground() 🚔
- fake.emoji_transport_sign() 🚮
- fake.emoji_transport_water() ⚓
- fake.emoji_travel_places() 🌒
- fake.emoji_warning() ⛔
- fake.emoji_writing() ✒
- fake.emoji_zodiac() ♐

### CODE REVIEW

**Comment by joke2k** [comment]
Should we merge this in the package or make it available as community provider?

File: faker/providers/emoji/__init__.py [hunk 1]
```diff
@@ -0,0 +1,4429 @@
...
+                [
+                    "U+1F507",
+                    "muted speaker",
+                    ["mute", "quiet", "silent", "speaker"],
+                ],
+                ["U+1F508", "speaker low volume", ["soft"]],
+                ["U+1F509", "speaker medium volume", ["medium"]],
+                ["U+1F50A", "speaker high volume", ["loud"]],
+                ["U+1F4E2", "loudspeaker", ["loud", "public address"]],
+                ["U+1F4
```
Comment (fcurella) [inline faker/providers/emoji/__init__.py]: Instead of creating the methods on `__new___`, would something in `__getattr__` work?

**Comment by fcurella** [comment]
@joke2k I think the needs to test code against emojis nowadays is common enough that this should be part of `faker`. I'm not sure that we want the category-specific methods. They're so many.

**Comment by Artmorse** [comment]
@joke2k When the PR #1706 will be merged. 
I'll do this one! You can assign me on it :smiley:

**Pull request closed without merging**